Source/WTF/wtf/NativePromise.h

```diff
@@ -593,27 +600,33 @@ class NativePromise final : public NativePromiseBase, public ConvertibleToNative
 
 public:
     template <class Dispatcher>
-    static Ref<AllPromiseType> all(Dispatcher& targetQueue, Vector<Ref<NativePromise>>& promises)
+    static Ref<AllPromiseType> all(Dispatcher& targetQueue, const Vector<Ref<NativePromise>>& promises)
     {
         static_assert(LooksLikeRCSerialDispatcher<typename RemoveSmartPointer<Dispatcher>::type>::value, "Must be used with a RefCounted SerialFunctionDispatcher");
-        if (promises.isEmpty())
-            return AllPromiseType::createAndResolve(Vector<ResolveValueType>());
-
+        if (promises.isEmpty()) {
+            if constexpr (std::is_void_v<ResolveValueT>)
+                return AllPromiseType::createAndResolve();
```

Comment:
I wonder whether we could avoid some of these constexpr if checks by having AllPromiseType::ResolveValueType be std::nullptr_t in case of void.
As mentioned by @squelart, we are sometimes using nullptr for void and sometimes special casing things.

Reply:
no, nullptr is only used internally to pass an (unused) argument to a method expecting one. It's not used outside of NativePromise internal

Reply:
Returning a `Vector<nullptr_t>` seems weird to me, I prefer having AllPromise::ResolveValueType to be `void`